ecl/hqlcpp/hqlcset.cpp

```diff
@@ -1424,24 +1415,24 @@ void GeneralSetCursor::buildAssignSelect(BuildCtx & ctx, const CHqlBoundTarget &
     {
         OwnedHqlExpr select = createSelectExpr(LINK(dsIndexExpr), LINK(element));
         if (!cursor->isConditional())
-            translator.buildExprAssign(ctx, target, select);
+            translator.buildExprOrAssign(ctx, target, select, tgt);
         else
         {
             // if (row) tgt = x else tgt = dft;
             BuildCtx subctx(ctx);
```

Comment:
@shamser I don't think it is possible to provoke it - even if the cursor might be conditional, a temporary set is then created avoiding the issue.

So please change this to 

    assertex(!cursor->isConditional());

and then delete the if and else branch  code.